### PR TITLE
Add XmlDoc analyzer and corresponding quick fixes

### PR DESCRIPTION
**Pull request by DedSec256**
- [x] Update FCS

### CODE REVIEW

File: ReSharper.FSharp/test/data/features/quickFixes/removeXmlDoc/XmlDoc 02.fs.gold [hunk 1]
```diff
@@ -0,0 +1,2 @@
+﻿() {caret}
```
Comment (auduchinok) [inline ReSharper.FSharp/test/data/features/quickFixes/removeXmlDoc/XmlDoc 02.fs.gold:1]: We should remove the space.

Please also consider this case in which keeping a whitespace line would be bad:

```fsharp
do
    /// 
    ()
```

File: ReSharper.FSharp/test/data/parsing/Type member - Member - XmlDoc 02.fs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+type A =
+    /// B
+    member x.B 
+        with get() = 5
+        and set (_: int) = ()
```
Comment (auduchinok) [inline ReSharper.FSharp/test/data/parsing/Type member - Member - XmlDoc 02.fs:5]: Do we need the accessors in this XmlDoc test?

File: ReSharper.FSharp/test/data/parsing/signatures/Type decl group - XmlDoc 04.fsi [hunk 3]
```diff
@@ -0,0 +1,4 @@
+type A
+
+/// B
+type B = 'T -> unit
```
Comment (auduchinok) [inline ReSharper.FSharp/test/data/parsing/signatures/Type decl group - XmlDoc 04.fsi:4]: What does function type usage test here?

File: ReSharper.FSharp/test/data/parsing/signatures/Type member - XmlDoc 03 - Constructor.fsi [hunk 4]
```diff
@@ -0,0 +1,6 @@
+module Module
+
+type T =
+
+    /// C
+    new: () -> T
```
Comment (auduchinok) [inline ReSharper.FSharp/test/data/parsing/signatures/Type member - XmlDoc 03 - Constructor.fsi:6]: Is it correct syntax at all?

**Review by auduchinok — CHANGES_REQUESTED** [review]
The 'remove' quick fix should cleanup whitespaces, like other features do.

The parser tests contain extra details that are not related to XmlDoc being attached to a particular declaration kind, please simplify them.

File: ReSharper.FSharp/test/data/parsing/Type member - Member - XmlDoc 02.fs (hunk 2 above)
Comment (DedSec256) [reply ReSharper.FSharp/test/data/parsing/Type member - Member - XmlDoc 02.fs:5]: Yes, since the problem was in properties with accessors
https://github.com/dotnet/fsharp/pull/12783

File: ReSharper.FSharp/test/data/parsing/signatures/Type member - XmlDoc 03 - Constructor.fsi (hunk 4 above)
Comment (DedSec256) [reply ReSharper.FSharp/test/data/parsing/signatures/Type member - XmlDoc 03 - Constructor.fsi:6]: Oh, thanks, there must be a `unit` instead of `()`

File: ReSharper.FSharp/test/data/parsing/Type member - Member - XmlDoc 02.fs (hunk 2 above)
Comment (auduchinok) [reply ReSharper.FSharp/test/data/parsing/Type member - Member - XmlDoc 02.fs:5]: So it requires two accessors with and a type annotation?

Comment (DedSec256) [reply ReSharper.FSharp/test/data/parsing/Type member - Member - XmlDoc 02.fs:5]: I think one getter will be enough

File: ReSharper.FSharp/test/data/features/quickFixes/removeXmlDoc/XmlDoc 03 - Remove line.fs.gold [hunk 5]
```diff
@@ -0,0 +1,3 @@
+﻿do
+{caret}
```
Comment (auduchinok) [inline ReSharper.FSharp/test/data/features/quickFixes/removeXmlDoc/XmlDoc 03 - Remove line.fs.gold:2]: Why the empty line is kept? The test name suggests it should be removed.

File: ReSharper.FSharp/test/data/features/quickFixes/removeXmlDoc/XmlDoc 02.fs [hunk 6]
```diff
@@ -1,2 +1,2 @@
-() /// Uwu{caret}
+()/// Uwu{caret}
```
Comment (auduchinok) [inline ReSharper.FSharp/test/data/features/quickFixes/removeXmlDoc/XmlDoc 02.fs:1]: Why remove the space in the test data? The test should show that quick fix removes it properly, shouldn't it?

Comment (DedSec256) [reply ReSharper.FSharp/test/data/features/quickFixes/removeXmlDoc/XmlDoc 02.fs:1]: I wanted to separately check the edge case when there is nothing to additionally delete between the comment and the node to the left. The `XmlDoc 03 - Remove line` test checks to remove spaces and newline to the left of the comment. 
Perhaps, we should separately check the removal of spaces between the comment and the node to the left, but it seemed redundant to me.

**Review by auduchinok — APPROVED** [review]

**Comment by auduchinok** [comment]
@DedSec256 Nice work!
